### PR TITLE
Add email validation on iteration completed page

### LINKED ISSUE #92: Add validation to email field on iteration completed page
The user email form currently accepts any input. We should add client side validation to the email field that validates the input before sending the request to the server.

### PR DESCRIPTION
**Pull request by musayann**
Added the email validation using html forms and added a validations
rule in user model

Closes #92

### CODE REVIEW

File: app/javascript/component/form/UserEmail.vue [hunk 1]
```diff
@@ -9,24 +9,41 @@
...
           </div>
-          <div class="w-30 bg-primary rounded-lg overflow-hidden">
-            <button
-              class="w-100 btn btn-primary py-2 text-base"
-              @click="save"
-            >
-              Submit
-            </button>
+          <div
+            v-if="form.errors.has('email')"
```
Comment (bumi) [inline app/javascript/component/form/UserEmail.vue]: can we somehow simply display all errors? - maybe that's a component?

File: app/models/user.rb [hunk 2]
```diff
@@ -8,6 +8,7 @@ class User < ApplicationRecord
   # Validations for the model
   validates :slug, presence: true, uniqueness: true
   validates :email, uniqueness: { allow_blank: true }
+  validates :email, format: { with: URI::MailTo::EMAIL_REGEXP, allow_blank: true }
```
Comment (bumi) [inline app/models/user.rb]: :+1:

File: app/javascript/component/form/UserEmail.vue (hunk 1 above)
Comment (musayann) [reply app/javascript/component/form/UserEmail.vue]: Yeah let me add that

Comment (musayann) [reply app/javascript/component/form/UserEmail.vue]: You mean displaying all errors at the same time?

Comment (bumi) [reply app/javascript/component/form/UserEmail.vue]: yes, as the reason for why the user could not be saved might not only be the email but also other validations.

**Pull request merged**